nuget/Microsoft.CodeAnalysis.Analyzers/Microsoft.CodeAnalysis.Analyzers.Package.csproj

```diff
@@ -10,11 +10,17 @@
     <ReleaseNotes>Diagnostic analyzers for the Microsoft .NET Compiler Platform (Roslyn)</ReleaseNotes>
     <PackageTags>Roslyn CodeAnalysis Compiler CSharp VB VisualBasic Diagnostic Analyzers Syntax Semantics</PackageTags>
 
-    <!-- 
-      PackageId is used by Restore. If we set it to Microsoft.CodeAnalysis.Analyzers 
-      Restore would conclude that there is a cyclic dependency between Microsoft.CodeAnalysis and Microsoft.CodeAnalysis.Analyzers.
+    <!--
+      PackageId is used by Restore. If we set it to Microsoft.CodeAnalysis.Analyzers,
+      Restore would conclude that there is a cyclic dependency between Microsoft.CodeAnalysis and
+      Microsoft.CodeAnalysis.Analyzers.
+
+      Additionally, for Unix, convert the paths to Windows-style paths. The project.assets.json
+      file uses '/' as a delimiter between the package id and the version. If keep the
+      forward-slashes in the path on Unix, the code parsing projects.assets.json gets confused
+      about where the package name ends and where the version begins.
     -->
-    <PackageId>*$(MSBuildProjectFullPath)*</PackageId>
+    <PackageId>*$(MSBuildProjectFullPath.Replace('/', '\\'))*</PackageId>
```

Comment:
❗ Just use the project file full name (including the extension). This is simpler, shorter, and avoids the need to explain platform-specific workarounds.

```xml
<PackageId>*$(MSBuildProjectFile)*</PackageId>
```

Reply:
There's a comment describing why the repo did it this way--@sharwell why doesn't it apply (anymore)?

https://github.com/dotnet/roslyn-analyzers/blob/0919dbcf9bb03874aec10ec190c2f7f2397c10fb/nuget/Microsoft.CodeAnalysis.Analyzers/Microsoft.CodeAnalysis.Analyzers.Package.csproj#L13-L17

Reply:
I think keeping the file name (including `.csproj`) still keeps the comment valid?

Reply:
I see, I think I misunderstood "The package ID here doesn't matter" 😛. It obviously does matter because it lets us use this workaround, it just doesn't need to match anything else.

Reply:
The package ID only matters in the sense that it meets both of the following two conditions:

1. It is not the same as the package ID for any other project in the solution.
2. It is not identical to the project file name _without_ extension. Even if we didn't use the file name with extension, the leading and trailing `*` characters would meet this condition.